chapters/object_types/samplers.txt

```diff
@@ -142,18 +142,24 @@ in column-major memory layout and are applied to column vectors
 
 Extension: `KHR_SAMPLER_PRIMITIVE`
 
-The primitive sampler samples an `Array1D` at unnormalized integer
-coordinates based on the `primitiveId` surface attribute, shifted by
-non-negative `offset`.
+The primitive sampler samples an `Array1D` at integer
+coordinates based on the `primitiveId` surface attribute.
+The `offset` parameter value is added to `primitiveID` before sampling.
 
 .Parameters accepted by all `primitive` samplers.
 [cols="<,<,<2",options="header,unbreakable"]
 |===================================================================================================
-| Name  | Type      | Description
-| array | `ARRAY1D` | backing array of the sampler
+| Name  | Type      | Default  | Description
+| array | `ARRAY1D` of <<Color>> |  | backing array of the sampler
+| offset | `UINT64` | 0 | offset into the array
 |===================================================================================================
 
-The `array` size must be at most <<Devices, device limit `geometryMaxIndex`>>.
```

Comment:
I removed this restriction since it becomes less clear why `geometryMaxIndex` would be a required boundary for this. Especially with the offset.

Reply:
I'm not sure if this has to be addressed here. I'd probably put some language in the section about arrays what happens if device internal operation causes out of bounds accesses.
In similar cases other khronos APIs say these accesses are undefined behavior and then there is some khr_robustness extension specifying behavior for those cases.

With ANARI being a bit higher level I'd probably just specify that it is safe in the "doesn't crash" sense but undefined in a what gets sampled/read sense.

Reply:
See for example https://registry.khronos.org/OpenGL/extensions/KHR/KHR_robustness.txt

> Variables of the same type ... Undefined behavior results from indexing
    an array with a non-constant expression that is greater than or equal to
    the array size or less than 0. If robust buffer access is enabled (see
    section 10.3.1.1rob of the OpenGL ES 3.1 API Specification), such
    indexing must not result in abnormal program termination. The results
    are still undefined, but implementations are encouraged to produce zero
    values for such accesses. Arrays only have a single dimension ...